cc/transport/grpc_streaming_transport.h

```diff
@@ -17,17 +17,102 @@
 #ifndef CC_TRANSPORT_GRPC_STREAMING_TRANSPORT_H_
 #define CC_TRANSPORT_GRPC_STREAMING_TRANSPORT_H_
 
+#include <grpc/grpc.h>
+#include <grpcpp/channel.h>
+#include <grpcpp/client_context.h>
+#include <grpcpp/create_channel.h>
+#include <grpcpp/grpcpp.h>
+
+#include <memory>
+
 #include "absl/status/statusor.h"
 #include "absl/strings/string_view.h"
 #include "cc/transport/transport.h"
 #include "oak_remote_attestation/proto/v1/messages.pb.h"
+#include "oak_remote_attestation/proto/v1/service_streaming.grpc.pb.h"
+#include "oak_remote_attestation/proto/v1/service_streaming.pb.h"
 
 namespace oak::transport {
 
 class GrpcStreamingTransport : public TransportWrapper {
  public:
-  absl::StatusOr<::oak::session::v1::AttestationBundle> GetEvidence() override;
-  absl::StatusOr<std::string> Invoke(absl::string_view request_bytes) override;
+  explicit GrpcStreamingTransport(
+      std::unique_ptr<::grpc::ClientReaderWriter<::oak::session::v1::RequestWrapper,
+                                                 ::oak::session::v1::ResponseWrapper>>
+          channel_reader_writer)
+      : channel_reader_writer_(std::move(channel_reader_writer)) {}
+
+  absl::StatusOr<::oak::session::v1::AttestationBundle> GetEvidence() override {
+    // Create request.
+    ::oak::session::v1::RequestWrapper request;
+    ::oak::session::v1::GetPublicKeyRequest get_public_key_request;
+    *request.mutable_get_public_key_request() = get_public_key_request;
+
+    // Send request.
+    auto response = Send(request);
+    if (!response.ok()) {
+      return response.status();
+    }
+
+    // Process response.
+    switch (response->response_case()) {
+      case ::oak::session::v1::ResponseWrapper::kGetPublicKeyResponseFieldNumber:
+        return response->get_public_key_response().attestation_bundle();
+      case ::oak::session::v1::ResponseWrapper::kInvokeResponseFieldNumber:
+        return absl::InternalError("received InvokeResponse instead of GetPublicKeyResponse");
+      case ::oak::session::v1::ResponseWrapper::RESPONSE_NOT_SET:
+        return absl::InternalError("received empty response");
+      default:
+        return absl::InternalError("received unsupported response type");
```

Comment:
I would just merge all of these in the default case, and print out the response for debugging.

Reply:
Done